internal/service/contentprovider/urlorlocalfile.go

```diff
@@ -0,0 +1,76 @@
+package contentprovider
+
+import (
+	"fmt"
+	"net/url"
+
+	commonerrors "github.com/kyma-project/modulectl/internal/common/errors"
+)
+
+// UrlOrLocalFile is a struct that represents eithera URL or a local file path.
+//
+//nolint:recvcheck // This is a value type, not a pointer type.
+type UrlOrLocalFile struct {
+	value string
+	url   *url.URL
+}
+
+// MustUrlOrLocalFile is a helper function that parses a string into a UrlOrLocalFile. Use only in tests!
+func MustUrlOrLocalFile(val string) UrlOrLocalFile {
+	var res UrlOrLocalFile
+	if err := res.FromString(val); err != nil {
+		panic(fmt.Sprintf("failed to parse UrlOrLocalFile: %s, error: %v", val, err))
+	}
+	return res
+}
+
+func (u UrlOrLocalFile) IsURL() bool {
+	return u.url != nil && len(u.url.Scheme) > 0 && len(u.url.Host) > 0
```

Comment:
Maybe worth a comment. I spent quite some time comparing this against the check in FromString. Maybe it would also be easier to just leave u.url nil if len(parsedURL.Scheme) == 0

Reply:
Thanks. Indeed the logic here should be moved into "FromString", and the checks in `IsURL` should be trivial.
I've modified the code trying to be more explicit. Please take a look if it's better now.